packages/cli/src/rpc/navie/register.ts

```diff
@@ -0,0 +1,62 @@
+import { warn } from 'console';
+import { NavieRpc } from '@appland/rpc';
+import {
+  AI,
+  ConversationThread,
+  ModelParameters,
+  ProjectDirectory,
+  ProjectParameters,
+} from '@appland/client';
+
+import { RpcHandler } from '../rpc';
+import { getLLMConfiguration } from '../llmConfiguration';
+import detectAIEnvVar from '../../cmds/index/aiEnvVar';
+import configuration from '../configuration';
+
+export async function register(
+  codeEditor: string | undefined
+): Promise<NavieRpc.V1.Register.Response> {
+  const modelParameters = {
+    ...getLLMConfiguration(),
+  } as ModelParameters;
+  const aiKeyName = detectAIEnvVar();
+  if (aiKeyName) modelParameters.aiKeyName = aiKeyName;
+
+  const configurationDirectories = await configuration().appmapDirectories();
+  const directories = configurationDirectories.map((dir) => {
+    const result: ProjectDirectory = {
+      hasAppMapConfig: dir.appmapConfig !== undefined,
+    };
+    if (dir.appmapConfig?.language) {
+      result.language = dir.appmapConfig.language;
+    }
+    return result;
+  });
+
+  const projectParameters: ProjectParameters = {
+    directoryCount: configurationDirectories.length,
+    directories,
+  };
+  if (codeEditor) projectParameters.codeEditor = codeEditor;
+
+  let thread: ConversationThread | undefined;
+  try {
+    thread = await AI.createConversationThread({ modelParameters, projectParameters });
+  } catch (err) {
+    warn(`Failed to create conversation thread`);
```

Comment:
TODO: The frontend may be expecting this to be an error, rather than a result where the thread is undefined. Maybe it would be better to update the Response type such that the thread is cannot be undefined, and raise an error here if the conversation can't be enrolled.